### PR TITLE
Add an endpoint for fetching projects for linked gcp account

### PR DESCRIPTION
**Pull request by siggisim**
This will allow the user to pick which project they'd like to deploy into

### CODE REVIEW

File: enterprise/server/gcplink/gcplink.go [hunk 1]
```diff
@@ -146,3 +149,106 @@ func makeTokenExchangeRequest(refreshToken string) (string, error) {
 type accessTokenResponse struct {
 	AccessToken string `json:"access_token"`
 }
+
+func GetGCPProject(env environment.Env, ctx context.Context, request *gcpb.GetGCPProjectRequest) (*gcpb.GetGCPProjectResponse, error) {
+	groupID := ""
+	if reqCtx := requestcontext.ProtoRequestContextFromContext(ctx); reqCtx != nil {
+		groupID = reqCtx.GetGroupId()
+	}
```
Comment (tylerwilliams) [inline enterprise/server/gcplink/gcplink.go:157]: is there a reason this isn't GetGroupID()? is it because these requests ignore the group picker

**Review by tylerwilliams — APPROVED** [review]

File: enterprise/server/gcplink/gcplink.go (hunk 1 above)
Comment (siggisim) [reply enterprise/server/gcplink/gcplink.go:157]: Na, I think I just cribbed it from the wrong place - will fix.

File: enterprise/server/gcplink/gcplink.go [hunk 2]
```diff
@@ -146,3 +149,106 @@ func makeTokenExchangeRequest(refreshToken string) (string, error) {
...
+
+	secretService := env.GetSecretService()
+	if secretService == nil {
+		return nil, status.FailedPreconditionError("secret service not available")
+	}
+	envVars, err := secretService.GetSecretEnvVars(ctx, groupID)
+	if err != nil {
+		return nil, err
+	}
+	envVars, err = ExchangeRefreshTokenForAuthToken(ctx, envVars, true)
```
[the remark this answers is not in the document]
Comment (siggisim) [reply enterprise/server/gcplink/gcplink.go:167]: Agreed, but I feel like the right way to do that would be to add a method to the SecretService to allow you to just fetch a single environment variable - but that felt a little out of the scope of this PR. I've added a TODO

File: enterprise/server/gcplink/gcplink.go (hunk 1 above)
Comment (siggisim) [reply enterprise/server/gcplink/gcplink.go]: Fixed

**Pull request merged**